WooCommerce/Classes/POS/TabBar/POSTabCoordinator.swift

```diff
@@ -143,7 +142,36 @@ private extension POSTabCoordinator {
     func presentPOSView(siteID: Int64) {
         Task { @MainActor [weak self] in
             guard let self else { return }
-            let serviceAdaptor = POSServiceLocatorAdaptor()
+
+            // Check local catalog eligibility before initializing infrastructure
+            // Try to use pre-created service from eligibility checker, otherwise create it now
+            let eligibilityService: POSLocalCatalogEligibilityServiceProtocol
+            if let preCreatedService = (eligibilityChecker as? POSTabEligibilityChecker)?.localCatalogEligibilityService {
+                eligibilityService = preCreatedService
+            } else {
+                // Fallback: assume we're POS eligible and create service
+                eligibilityService = await POSLocalCatalogEligibilityService(
+                    siteID: siteID,
+                    catalogSizeChecker: POSCatalogSizeChecker(credentials: credentials,
+                                                              selectedSite: defaultSitePublisher,
+                                                              appPasswordSupportState: isAppPasswordSupported
+                    ),
+                    posTabEligibilityState: .eligible
+                )
+            }
+
+            switch eligibilityService.eligibilityState {
+            case .ineligible(reason: .catalogSizeCheckFailed):
+                // If we cached a failed check, we can recover by refreshing the value before we next open POS
+                await eligibilityService.refreshEligibilityState()
```

Comment:
As I understand, we will get a refreshed state in 3 other cases:
- After the login
- After changing the site
- Fresh launch of the app

If POS and the app are continued to be used, we're not checking eligibility. If we introduce a remote feature flag, for example, we may want to know earlier if the site is still eligible.

One of the potential places is `ForegroundPOSCatalogSyncDispatcher`. We not only need to introduce `eligibilityState` checks into the dispatcher, but it could also refresh the state before checking. We could make sure a task is created to do that. WDYT?